amazon_s3_download_block.py

```diff
@@ -1,22 +1,47 @@
-from nio.properties import (VersionProperty, StringProperty,
+import boto3
+
+from nio.block.base import Block
+from nio.signal.base import Signal
+from nio.properties import (VersionProperty, StringProperty, PropertyHolder,
                             ObjectProperty)
-from .amazon_base_block import AmazonBase, AWSCreds
-from nio.util.discovery import discoverable
 
 
-@discoverable
-class AmazonS3(AmazonBase):
+class AWSCreds(PropertyHolder):
+    aws_access_key_id = StringProperty(
+        title="Access Key ID", default="", allow_none=False)
+    aws_secret_access_key = StringProperty(
+        title="Secret Access Key", default="", allow_none=False)
+    aws_session_token = StringProperty(
+        title="Session Token", default="", allow_none=False)
```

Comment:
Is this property required? I seem to remember using amazon stuff with just the first two. Maybe `allow_none = True` on this one if that's the case?